### DISCUSSION TITLE
Error: (0 , react_hook_form__WEBPACK_IMPORTED_MODULE_3__.useForm) is not a function

### DISCUSSION THREAD

**Discussion by HugoQueva** [react-hook-form/react-hook-form#11278] [Q&A] [answered]
Hi, I'm using `"@hookform/resolvers": "^3.3.2"` and `"react-hook-form": "^7.49.0",`.
When I'm trying to initiate a form with `useForm`, it throw me an error saying `useForm` is not a function.
I have tried to uninstall  the **node_modules** directory and use earlier versions. None of these worked.

I saw a man with similar issue, but the solution didn't worked for me.
 
![image](https://github.com/react-hook-form/react-hook-form/assets/139586455/2226d335-a9c7-442e-8bc5-54b220840e6a)
![image](https://github.com/react-hook-form/react-hook-form/assets/139586455/546d5b26-5330-4e03-8ca4-59f90aafe183)

**Comment by maruf-abd** [comment, marked as answer]
@HugoQuevaReborn 
Please check that you have included "use client"; on the top or not.

**Reply by Levi9111** [reply]
I didn't. and now the solution worked for me. Thank you

**Reply by cristovaom** [reply]
on the top , look import or not?

**Reply by craigmcc** [reply]
The "use client" directive needs to go above any imports.  The Next.JS documentation says you can have comments before it, but I find it easier to just always put this at the very top of my client components.

**Reply by cristovaom** [reply]
Ah okay , worked for me , thx

**Reply by dushmanta05** [reply]
Thanks for the help it worked.

**Reply by pvtdao** [reply]
It worked. Thank you

**Reply by yogithesymbian** [reply]
i dont know why but u saved my life
i use `"use_client"` ahhhh its "use client" without underscore ..  . thankyou

**Reply by kflieder** [reply]
YOOOOOOOOOOOOO, this is how i know AI isnt going to take over software engineering anytime soon, because chatgpt did not pick up on the lack of "use client" ahahah thank youuuuuu humans

**Reply by nileshparwan** [reply]
thx, works also for me !

**Reply by Tahiraawan** [reply]
OMG it works

**Reply by Bomote** [reply]
Can someone please explain why "use client" worked for this?

**Reply by DivyarajsinhRana** [reply]
> Can someone please explain why "use client" worked for this?

As hooks can not work on server components , need to declare it as 'use client'.

**Reply by monu-shaw** [reply]
But I want render server side as by default "use server". As all already render example code Image
![Screenshot from 2024-07-02 21-09-11](https://github.com/react-hook-form/react-hook-form/assets/68293997/56b9783d-c2d2-42ad-9171-dbdbd48e8d03)
How I can do so.

**Reply by DivyarajsinhRana** [reply]
Make this SignUpUI  component as client component using 'use-client' and import it into one parent level component which will be default server component.

**Reply by monu-shaw** [reply]
Definitely I can do so , no issues but where I taken advantage of server side, as you can see in my code there is no client side components & hooks are used. And antDesign also support server side rendering.
Advantage servers sides in SEO, now gemini also use SEO to understand website or brand.
For eg. Like any one ask gemini "how start on platform xyz" or "what needed to register on xyz".
As because bot could not understand about client side components but they can understand about server side rendered components.

**Reply by DivyarajsinhRana** [reply]
As per my understanding of Next JS from it docs is that we should not use event handlers and hooks in server component , i can see you have used onClick and some hooks in your code.

**Reply by monu-shaw** [reply]
Yes, I also know this but as my code get stuck error on line 10, so I didn't look there definitely I will remove that

**Reply by illustrofia** [reply]
Thank you for having patience with us, @maruf-abd  🙏🏻 🤗

**Comment by sp-rahul** [comment]
use client

Then It will work
